.github/workflows/scale-test-egw.yaml

```diff
@@ -0,0 +1,485 @@
+          kubectl rollout status ds/aws-node -n kube-system -w
+
+      - name: Install Cilium CLI
+        uses: cilium/cilium-cli@99ff3d9edd4050d7b5c9d1f51dcf6ebeeb613d15 # v0.16.15
+        with:
+          skip-build: ${{ env.CILIUM_CLI_SKIP_BUILD }}
+          image-repo: ${{ env.CILIUM_CLI_IMAGE_REPO }}
+          image-tag: ${{ env.CILIUM_CLI_VERSION }}
+
+      - name: Install Cilium
+        run: |
+          cilium install --dry-run-helm-values ${{ steps.vars.outputs.cilium_install_defaults }}
+          cilium install ${{ steps.vars.outputs.cilium_install_defaults }}
+
+      - name: Wait for Cilium status to be ready
+        run: |
+          cilium status --wait
+
+      - name: Check that AWS leftover iptables chains have been removed
+        run: |
+          for pod in $(kubectl get po -n kube-system -l app.kubernetes.io/name=cilium-agent -o name); do
+            echo "Checking ${pod}"
+            if kubectl exec -n kube-system  ${pod} -c cilium-agent -- iptables-save | grep --silent ':AWS'; then
+              echo "Unexpected AWS leftover iptables chains"
+              kubectl exec -n kube-system ds/cilium -- iptables-save | grep ':AWS'
+              exit 1
+            fi
+          done
+
+      - name: Run preflight steps
+        shell: bash
+        working-directory: ./scaffolding/egw-scale-utils/test
+        env:
+          EGW_IMAGE_TAG: ${{ env.egw_utils_ref }}
+        run: |
+          if [[ "${{ matrix.test_type }}" == "md-bs" ]]; then
+            ./preflight.sh baseline
+          else
+            ./preflight.sh
+          fi
+
+          cat ./manifests/*
+
+      - name: Run CL2
+        id: run-cl2
+        working-directory: ./perf-tests/clusterloader2
+        shell: bash
+        timeout-minutes: 40
+        env:
+          CL2_PROMETHEUS_PVC_ENABLED: "false"
+          CL2_ENABLE_PVS: "false"
+          CL2_PROMETHEUS_SCRAPE_CILIUM_OPERATOR: "true"
+          CL2_PROMETHEUS_SCRAPE_CILIUM_AGENT: "true"
+          CL2_PROMETHEUS_MEMORY_SCALE_FACTOR: "2.0"
+          CL2_PROMETHEUS_NODE_SELECTOR: 'role.scaffolding/monitoring: "true"'
+          CL2_NUM_EGW_CLIENTS: "${{ steps.vars.outputs.num_client_pods }}"
+          CL2_EGW_CLIENTS_QPS: "${{ steps.vars.outputs.client_qps }}"
+        run: |
+          mkdir ./report
+
+          # CL2 hardcodes module paths to live in next to the config, even
+          # if the path given is relative.
+          cp ../../.github/actions/cl2-modules/cilium-agent-pprofs.yaml ../../scaffolding/egw-scale-utils/test/
+          cp ../../.github/actions/cl2-modules/cilium-metrics.yaml ../../scaffolding/egw-scale-utils/test/
+          echo \
+            '{"CL2_ADDITIONAL_MEASUREMENT_MODULES": ["./cilium-agent-pprofs.yaml", "./cilium-metrics.yaml"]}' \
+            > modules.yaml
+
+          go run ./cmd/clusterloader.go \
+            -v=2 \
+            --testconfig=../../scaffolding/egw-scale-utils/test/config.yaml \
```

Comment:
Is there a reason why the test lives in the scaffolding repo?
Most of the tests that we have are right now in .github/actions/cl2-modules
which also allows to reuse the same modules easier, for example:
```
- module:
    path: ../cilium-agent-pprofs.yaml
    params:
      action: gather
```

Reply:
Yes, having it all under the same roof makes it simpler to develop and test. I added some scripts for testing on kind that lives [here](https://github.com/cilium/scaffolding/tree/main/scenarios/egw-masq-delay), and they import the config that lives [here](https://github.com/cilium/scaffolding/blob/main/egw-scale-utils/test/config.yaml). To handle importing the modules, the config uses the same pattern as the upstream load test and takes a [ADDITIONAL_MEASUREMENT_MODULES](https://github.com/cilium/scaffolding/blob/d0e4d60cf847b37fbde8e9ed4a11c70d1ae349ef/egw-scale-utils/test/config.yaml#L3) parameter.

Reply:
I can move it into `cilium/cilium` though if that's the pattern that we are looking to use.

Reply:
Moved everything to `cilium/cilium` under `.github/actions/cl2-modules/egw`. The only thing left in the scaffolding repository is the source for the `egw-scale-utils` docker image. We can follow up with a PR to scaffolding to remove the stuff that was moved over here.